backup/queries_functions.go

```diff
@@ -831,13 +831,37 @@ func (e Extension) FQN() string {
 func GetExtensions(connectionPool *dbconn.DBConn) []Extension {
 	results := make([]Extension, 0)
 
-	query := fmt.Sprintf(`
-	SELECT e.oid,
-		quote_ident(extname) AS name,
-		quote_ident(n.nspname) AS schema
-	FROM pg_extension e
-		JOIN pg_namespace n ON e.extnamespace = n.oid
-	WHERE e.oid >= %d`, FIRST_NORMAL_OBJECT_ID)
+	query := ""
+	if connectionPool.Version.Before("6") {
+		query = fmt.Sprintf(`
+		SELECT e.oid,
+			quote_ident(extname) AS name,
+			quote_ident(n.nspname) AS schema
+		FROM pg_extension e
+			JOIN pg_namespace n ON e.extnamespace = n.oid
+		WHERE e.oid >= %d
+		ORDER BY 1`, FIRST_NORMAL_OBJECT_ID)
+	} else {
+		query = fmt.Sprintf(`
+		WITH recursive cte AS (
```

Comment:
Not critical, but this query can be simplified. We can avoid double join of pg_extension, move oid filter inside cte, avoid duplicate filtering on pg_depend.
```
with recursive ext_tree as (
    select e.extname, e.extnamespace, e.oid as parent_oid, d.refobjid as child_oid
    from pg_catalog.pg_extension e
    left join pg_catalog.pg_depend d on d.objid = e.oid
                                     and d.classid = 'pg_catalog.pg_extension'::pg_catalog.regclass
                                     and d.refclassid = 'pg_catalog.pg_extension'::pg_catalog.regclass
    where e.oid >= 16384
),
cte as (
    select parent_oid as oid, 1 as level, extname, extnamespace  from ext_tree where child_oid is null
    union
    select e.parent_oid, cte.level + 1, e.extname, e.extnamespace from cte join ext_tree e on cte.oid = e.child_oid
)
select e.oid,
       quote_ident(e.extname) AS name,
       quote_ident(n.nspname) AS schema
from cte e
join pg_catalog.pg_namespace as n on e.extnamespace = n.oid
order by e.level;
```
(it's ok to have a second cte for recursive statement)

But this solution is still affected by duplicate rows bug mentioned by @red1452

Reply:
applied